test_runner/fixtures/neon_fixtures.py

```diff
@@ -2155,7 +2155,7 @@ def http_client(self, auth_token: Optional[str] = None) -> PageserverHttpClient:
 
     def assert_no_errors(self):
         logfile = open(os.path.join(self.env.repo_dir, "pageserver.log"), "r")
-        error_or_warn = re.compile("ERROR|WARN")
+        error_or_warn = re.compile(r"^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z)\s+(ERROR|WARN)")
```

Comment:
I'm a little worried that this is too strict, and we might not notice if it stops working because of some small change in the timestamp format.

How about just requiring a space before the ERROR or WANR, i.e. `\s(ERROR|WARN)`?

Reply:
What if we instead switch to json log format for python and have the parsing solved for us?
AFAIK, log level is passed as another field there.

Reply:
```suggestion
        error_or_warn = re.compile(r"\s(ERROR|WARN)")
```

Reply:
Whoops, didn't see the comment by @SomeoneToIgnore before committing that suggestion.

> Why not json

It really helps with the flaky test hunting to be able to read the log files in allure reports. I think the stricter would be fine here just as well, we could work around against introducing subtle bugs differently like expecing the first line to always match + make re to capture all levels, filter it in python to WARN, ERROR...

Reply:
Just to argue with the readability, current json format is not that bad.
But I do not insist on your approach, that's simpler indeed.

Log example:

```
{"timestamp":"2023-03-13T10:47:29.194734Z","level":"INFO","fields":{"message":"version: git:ce8fbbd910eabc1adf1982164f97bd8e2b108f8f-modified failpoints: true, features: [] launch_timestamp: 2023-03-13 10:47:29.190653 UTC"}}
{"timestamp":"2023-03-13T10:47:29.214211Z","level":"INFO","fields":{"message":"Claimed pid file at \"/Users/someonetoignore/work/neon/neon/.neon/pageserver.pid\""}}
{"timestamp":"2023-03-13T10:47:29.214286Z","level":"INFO","fields":{"message":"Starting pageserver http handler on 127.0.0.1:9898"}}
{"timestamp":"2023-03-13T10:47:29.214436Z","level":"INFO","fields":{"message":"Starting pageserver pg protocol handler on 127.0.0.1:64000"}}
{"timestamp":"2023-03-13T10:47:29.215589Z","level":"INFO","fields":{"message":"Initialized broker client with endpoints: http://127.0.0.1:50051/"}}
{"timestamp":"2023-03-13T10:47:29.215600Z","level":"INFO","fields":{"message":"Using auth: Trust"}}
{"timestamp":"2023-03-13T10:47:29.215604Z","level":"INFO","fields":{"message":"No JWT token for authentication with Safekeeper detected"}}
{"timestamp":"2023-03-13T10:47:29.216333Z","level":"INFO","fields":{"message":"tenant 98d670ab7bee6f0051494306a1ab888f is assumed to be loadable, starting load operation"},"span":{"name":"init_tenant_mgr"},"spans":[{"name":"init_tenant_mgr"}]}
{"timestamp":"2023-03-13T10:47:29.216420Z","level":"INFO","fields":{"message":"loading tenantconf from /Users/someonetoignore/work/neon/neon/.neon/tenants/98d670ab7bee6f0051494306a1ab888f/config"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"spawn_load"},"spans":[{"name":"init_tenant_mgr"},{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"spawn_load"}]}
{"timestamp":"2023-03-13T10:47:29.216876Z","level":"INFO","fields":{"message":"spawned load into background"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"spawn_load"},"spans":[{"name":"init_tenant_mgr"},{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"spawn_load"}]}
{"timestamp":"2023-03-13T10:47:29.217059Z","level":"INFO","fields":{"message":"loading tenant task"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.217161Z","level":"INFO","fields":{"message":"tenant 19cbf2bf51f42a5a5a90aa8954fb3e42 is assumed to be loadable, starting load operation"},"span":{"name":"init_tenant_mgr"},"spans":[{"name":"init_tenant_mgr"}]}
{"timestamp":"2023-03-13T10:47:29.217168Z","level":"INFO","fields":{"message":"loading tenantconf from /Users/someonetoignore/work/neon/neon/.neon/tenants/19cbf2bf51f42a5a5a90aa8954fb3e42/config"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"spawn_load"},"spans":[{"name":"init_tenant_mgr"},{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"spawn_load"}]}
{"timestamp":"2023-03-13T10:47:29.217441Z","level":"INFO","fields":{"message":"spawned load into background"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"spawn_load"},"spans":[{"name":"init_tenant_mgr"},{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"spawn_load"}]}
{"timestamp":"2023-03-13T10:47:29.217475Z","level":"INFO","fields":{"message":"loading tenant task"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.217503Z","level":"INFO","fields":{"message":"Processed 2 local tenants at startup"},"span":{"name":"init_tenant_mgr"},"spans":[{"name":"init_tenant_mgr"}]}
{"timestamp":"2023-03-13T10:47:29.217948Z","level":"INFO","fields":{"message":"Rebuilt layer map. Did 1 insertions to process a batch of 1 updates."},"span":{"timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"load_local_timeline"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},{"timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"load_local_timeline"}]}
{"timestamp":"2023-03-13T10:47:29.217962Z","level":"INFO","fields":{"message":"loaded layer map with 1 layers at 0/16B5A50, total physical size: 25821184"},"span":{"timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"load_local_timeline"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},{"timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"load_local_timeline"}]}
{"timestamp":"2023-03-13T10:47:29.217970Z","level":"INFO","fields":{"message":"spawning flush loop"},"span":{"timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"load_local_timeline"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},{"timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"load_local_timeline"}]}
{"timestamp":"2023-03-13T10:47:29.218032Z","level":"INFO","fields":{"message":"Rebuilt layer map. Did 1 insertions to process a batch of 1 updates."},"span":{"timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"load_local_timeline"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},{"timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"load_local_timeline"}]}
{"timestamp":"2023-03-13T10:47:29.218043Z","level":"INFO","fields":{"message":"loaded layer map with 1 layers at 0/16B5A50, total physical size: 25821184"},"span":{"timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"load_local_timeline"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},{"timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"load_local_timeline"}]}
{"timestamp":"2023-03-13T10:47:29.218049Z","level":"INFO","fields":{"message":"spawning flush loop"},"span":{"timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"load_local_timeline"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},{"timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"load_local_timeline"}]}
{"timestamp":"2023-03-13T10:47:29.218086Z","level":"INFO","fields":{"message":"Activating tenant 19cbf2bf51f42a5a5a90aa8954fb3e42"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.218096Z","level":"INFO","fields":{"message":"Activating tenant 98d670ab7bee6f0051494306a1ab888f"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.218107Z","level":"INFO","fields":{"message":"launching WAL receiver for timeline a6cfe8a1ca34765fd9a432d6193856b2 of tenant 19cbf2bf51f42a5a5a90aa8954fb3e42"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.218113Z","level":"INFO","fields":{"message":"launching WAL receiver for timeline a4273b74e519eb7f2d93b2c6245cd8ec of tenant 98d670ab7bee6f0051494306a1ab888f"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.218366Z","level":"INFO","fields":{"message":"starting"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"compaction_loop"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"compaction_loop"}]}
{"timestamp":"2023-03-13T10:47:29.218135Z","level":"INFO","fields":{"message":"Done"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.218383Z","level":"INFO","fields":{"message":"WAL receiver manager started, connecting to broker"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:29.218309Z","level":"INFO","fields":{"message":"started flush loop"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"layer flush task"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"layer flush task"}]}
{"timestamp":"2023-03-13T10:47:29.218328Z","level":"INFO","fields":{"message":"starting"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"compaction_loop"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"compaction_loop"}]}
{"timestamp":"2023-03-13T10:47:29.218393Z","level":"INFO","fields":{"message":"Done"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"load"}]}
{"timestamp":"2023-03-13T10:47:29.218497Z","level":"INFO","fields":{"message":"initial load for tenant 98d670ab7bee6f0051494306a1ab888f finished!"}}
{"timestamp":"2023-03-13T10:47:29.218411Z","level":"INFO","fields":{"message":"initial load for tenant 19cbf2bf51f42a5a5a90aa8954fb3e42 finished!"}}
{"timestamp":"2023-03-13T10:47:29.218431Z","level":"INFO","fields":{"message":"starting"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"gc_loop"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","name":"gc_loop"}]}
{"timestamp":"2023-03-13T10:47:29.218438Z","level":"INFO","fields":{"message":"WAL receiver manager started, connecting to broker"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:29.218288Z","level":"INFO","fields":{"message":"started flush loop"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"layer flush task"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"layer flush task"}]}
{"timestamp":"2023-03-13T10:47:29.218269Z","level":"INFO","fields":{"message":"starting"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"gc_loop"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","name":"gc_loop"}]}
{"timestamp":"2023-03-13T10:47:29.219202Z","level":"INFO","fields":{"message":"Attempt #1, failed to subscribe for timeline 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:29.219241Z","level":"INFO","fields":{"message":"Backoff: waiting 1.1 seconds before processing with the task"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:29.219294Z","level":"INFO","fields":{"message":"Attempt #1, failed to subscribe for timeline 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:29.219307Z","level":"INFO","fields":{"message":"Backoff: waiting 1.1 seconds before processing with the task"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:30.322711Z","level":"INFO","fields":{"message":"Attempt #2, failed to subscribe for timeline 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:30.322913Z","level":"INFO","fields":{"message":"Backoff: waiting 1.2100000000000002 seconds before processing with the task"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:30.323053Z","level":"INFO","fields":{"message":"Attempt #2, failed to subscribe for timeline 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:30.323097Z","level":"INFO","fields":{"message":"Backoff: waiting 1.2100000000000002 seconds before processing with the task"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:31.535961Z","level":"INFO","fields":{"message":"Attempt #3, failed to subscribe for timeline 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:31.536139Z","level":"INFO","fields":{"message":"Backoff: waiting 1.3310000000000004 seconds before processing with the task"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:31.536516Z","level":"INFO","fields":{"message":"Attempt #3, failed to subscribe for timeline 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:31.536558Z","level":"INFO","fields":{"message":"Backoff: waiting 1.3310000000000004 seconds before processing with the task"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:32.868806Z","level":"INFO","fields":{"message":"Attempt #4, failed to subscribe for timeline 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:32.868840Z","level":"INFO","fields":{"message":"Backoff: waiting 1.4641000000000004 seconds before processing with the task"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:32.868858Z","level":"INFO","fields":{"message":"Attempt #4, failed to subscribe for timeline 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:32.868876Z","level":"INFO","fields":{"message":"Backoff: waiting 1.4641000000000004 seconds before processing with the task"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:34.336359Z","level":"INFO","fields":{"message":"Attempt #5, failed to subscribe for timeline 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:34.336520Z","level":"INFO","fields":{"message":"Backoff: waiting 1.6105100000000006 seconds before processing with the task"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:34.336323Z","level":"INFO","fields":{"message":"Attempt #5, failed to subscribe for timeline 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec updates in broker: status: Unavailable, message: \"error trying to connect: tcp connect error: Connection refused (os error 61)\", details: [], metadata: MetadataMap { headers: {} }"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:34.336623Z","level":"INFO","fields":{"message":"Backoff: waiting 1.6105100000000006 seconds before processing with the task"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
^C{"timestamp":"2023-03-13T10:47:34.690052Z","level":"INFO","fields":{"message":"Got SIGINT. Terminating gracefully in fast shutdown mode"}}
{"timestamp":"2023-03-13T10:47:34.690120Z","level":"INFO","fields":{"message":"waiting for libpq endpoint listener to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690220Z","level":"INFO","fields":{"message":"waiting for walreceiver for timeline 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690237Z","level":"INFO","fields":{"message":"WAL receiver shutdown requested, shutting down"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:34.690247Z","level":"INFO","fields":{"message":"WAL receiver shutdown requested, shutting down"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"wal_connection_manager"}]}
{"timestamp":"2023-03-13T10:47:34.690282Z","level":"INFO","fields":{"message":"waiting for walreceiver for timeline 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690377Z","level":"INFO","fields":{"message":"waiting for http endpoint listener to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690550Z","level":"INFO","fields":{"message":"waiting for layer eviction for 98d670ab7bee6f0051494306a1ab888f/a4273b74e519eb7f2d93b2c6245cd8ec to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690575Z","level":"INFO","fields":{"message":"shutting down"},"span":{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"eviction_task"},"spans":[{"tenant_id":"98d670ab7bee6f0051494306a1ab888f","timeline_id":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"eviction_task"}]}
{"timestamp":"2023-03-13T10:47:34.690584Z","level":"INFO","fields":{"message":"eviction task finishing"}}
{"timestamp":"2023-03-13T10:47:34.690593Z","level":"INFO","fields":{"message":"shutting down layer flush task"},"span":{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"layer flush task"},"spans":[{"tenant":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline":"a6cfe8a1ca34765fd9a432d6193856b2","name":"layer flush task"}]}
{"timestamp":"2023-03-13T10:47:34.690608Z","level":"INFO","fields":{"message":"shutting down"},"span":{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"eviction_task"},"spans":[{"tenant_id":"19cbf2bf51f42a5a5a90aa8954fb3e42","timeline_id":"a6cfe8a1ca34765fd9a432d6193856b2","name":"eviction_task"}]}
{"timestamp":"2023-03-13T10:47:34.690638Z","level":"INFO","fields":{"message":"eviction task finishing"}}
{"timestamp":"2023-03-13T10:47:34.690618Z","level":"INFO","fields":{"message":"waiting for garbage collector for tenant 19cbf2bf51f42a5a5a90aa8954fb3e42 to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690654Z","level":"INFO","fields":{"message":"waiting for compactor for tenant 19cbf2bf51f42a5a5a90aa8954fb3e42 to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690608Z","level":"INFO","fields":{"message":"shutting down layer flush task"},"span":{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"layer flush task"},"spans":[{"tenant":"98d670ab7bee6f0051494306a1ab888f","timeline":"a4273b74e519eb7f2d93b2c6245cd8ec","name":"layer flush task"}]}
{"timestamp":"2023-03-13T10:47:34.690662Z","level":"INFO","fields":{"message":"waiting for layer flush task to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690672Z","level":"INFO","fields":{"message":"waiting for garbage collector for tenant 98d670ab7bee6f0051494306a1ab888f to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690677Z","level":"INFO","fields":{"message":"waiting for layer flush task to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690680Z","level":"INFO","fields":{"message":"waiting for layer eviction for 19cbf2bf51f42a5a5a90aa8954fb3e42/a6cfe8a1ca34765fd9a432d6193856b2 to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690684Z","level":"INFO","fields":{"message":"waiting for compactor for tenant 98d670ab7bee6f0051494306a1ab888f to shut down"}}
{"timestamp":"2023-03-13T10:47:34.690688Z","level":"INFO","fields":{"message":"Shut down successfully completed"}}
```

Reply:
my earlier:
> [...] I think the ~stricter~ strict re would be fine here just as well, we could work around against introducing subtle bugs differently like expecing the first line to always match + make re to capture all levels, filter it in python to WARN, ERROR...

This would be something like (patch): https://gist.github.com/koivunej/f21bb59621e46a48f22846a22aeab5b1

Reply:
Looking at the json, I would be able to make sense of it, withdrawing my opposition :)